### PR TITLE
re-order gateway support

### PR DESCRIPTION
**Pull request by jvonau**
http://paste.fedoraproject.org/301400/19992145/

### CODE REVIEW

**Comment by jvonau** [comment]
Always install all network options during the first pass, detected by the absence of xs_domain_name, then apply the detected network role anytime after.

**Comment by georgejhunt** [comment]
I'd like a script that resets the whole automatic recognition back to the beginning.  I'm aware of deleting ifcfg_*, sysconfig {lan,wan}, domainname, -- but you probably know of more that needs to be set back to a pristine state (maybe that should be the first step in xsce-network).

I think you disable quite a bit, and do a lot of speedup, if the domainname does not change. But you change the domainname at the beginning rather than the end of the process.

If the network goes down, or a pip install fails, during the next time you run xsce-network, "prepped" will be set, and I'll never be able to complete the rest of the network install.

**Comment by jvonau** [comment]
Ok, maybe I'm a bit obtuse, but 'Three times, I've encoundered pip errors during loader process regarding __dso_handle. This time I yum removed gcc-4.7, and "yum install compat-gcc-43".' kind of tells me there are issues in 3-base-server/xsce-admin before the network playbook is even reached. Would you be so kind as to share the xsce-install.log file so I can have a look also?

**Comment by georgejhunt** [comment]
I can confirm similar experience with gcc. I tried compat-gcc-43, but after
I had run install-console, the version had changed to 4.7.2. -- which did
not spew the __dso_handle errors -- I assumed that pip had requested the
uprade.

I don't think this log is helpful, because the zmq  creates
/lib/php/modules step reports "changed".  gcc --version = 4.7.2

**Comment by georgejhunt** [comment]
And I though that readonly-root might be messing me up. But I changed
TEMPORARY-STATE=no and rebooted.  But the symptoms did not change.

The pecl install zmq-beta is supposed to generate /lib/php/modules/zmq.so
-- but does not

**Comment by jvonau** [comment]
pecl is provided by php-pear, and s/b installed in the yum command above that call. pecl download zmq-beta does need to have internet access the first time around to retrieve the source but once it's on disk the next part at pecl install zmq*tgz installs it. Is the problem that the download is not present or is present but does not install? From other reports you have said that install-console has ran to completion on the XO-1.5 but now it's not?

**Comment by jvonau** [comment]
Think I see what might be going on Try:   yes '' | pecl install -f zmq*tgz 
Note the added -f

**Comment by jvonau** [comment]
Not sure why the current code works in some places and not in others but I can reproduce the problem by deleting zmq.so on a configured machine.

I've found the change below to be helpful, think "yes ''" was helping to mask the issue of needing to use  -f on any subsequent re-installs and I have no clue on why that is needed. 
```
diff --git a/roles/xsce-admin/tasks/cmdsrv.yml b/roles/xsce-admin/tasks/cmdsrv.yml
index 0972a63..8392647 100644
--- a/roles/xsce-admin/tasks/cmdsrv.yml
+++ b/roles/xsce-admin/tasks/cmdsrv.yml
@@ -40,8 +40,9 @@
     - download2
 
 - name: Install php binding for zeromq
-  shell: yes '' | pecl install zmq*tgz
-         creates="{{ phplib_dir }}/modules/zmq.so"
+  shell: pecl install -f zmq*tgz
+  args:
+    creates: "{{ phplib_dir }}/modules/zmq.so"
 
 - name: Download speedtest-cli
   pip: name=speedtest-cli
```
Just to make things more interesting there has been an update from upstream, via pip that might skew things a bit also:
Jul 24  2015 ../pip-packages/pyzmq-14.7.0.tar.gz on the F18 trimslice
Nov 27 01:28 ../pip-packages/pyzmq-15.1.0.tar.gz om the F22 liveiso

**Comment by tim-moody** [comment]
is this doing an offline install?  was download ever separated from install?

not sure version of python binding is relevant to absence of php binding .so, but if the api didn't work it would be.

**Comment by jvonau** [comment]
On Wed, Dec 16, 2015 at 10:02 AM, Tim Moody <notifications@github.com>
wrote:

> is this doing an offline install?
>
I've mentioned this before, one can fake that by setting use_cache &
no_network to True in local_vars.yml, but yes the F18 box is nothing but a
LanController with no net access. Once a single pass through
install-console is done the above overrides are effective in suppressing
the downloads.


> was download ever separated from install?
>
Yes, back in March so I'm at a bit of a loss as to why only now is this
bubbling to the surface.


> not sure version of python binding is relevant to absence of php binding
> .so, but if the api didn't work it would be.
>
Just trying to narrow down why there is a change in behavior now, but the
above change works with both F18 and F22 over here.

**Comment by georgejhunt** [comment]
Not sure this is relevant. I did a "pear config-show" and got a message
right at the top saying it could not open shared object
../php/modules/zmq.so.  So there is some memory/state in pear which we need
to clear out for rerunning our playbook.

When I try to run "pecl install -f zmq*beta" from the command line, I get

WARNING: channel "pecl.php.net" has updated its protocols, use "pecl
channel-update pecl.php.net" to update
downloading zmq-1.1.2.tgz ...
Starting to download zmq-1.1.2.tgz (39,573 bytes)
..........done: 39,573 bytes
12 source files, building
running: phpize
Configuring for:
PHP Api Version:         20100412
Zend Module Api No:      20100525
Zend Extension Api No:   220100525
Please provide the prefix of libzmq installation [autodetect] : building in
/var/tmp/pear-build-rootEzdFPW/zmq-1.1.2
running: /var/tmp/zmq/configure --with-zmq
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... configure: error: in
`/var/tmp/pear-build-rootEzdFPW/zmq-1.1.2':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
ERROR: `/var/tmp/zmq/configure --with-zmq' failed

the /var/tmp/pear-build disappears so it's hard to follow up. When I untar
the zmq*tgz, there's a config.m4, but no .configure, which I might be able
to run manually, and find out why it thinks we are cross-compiling.

**Comment by jvonau** [comment]
The -f does a forced reinstall, think that is the key here.

interesting, while MY F18 returns
```
[root@schoolserver xsce]# pear config-show
CONFIGURATION (CHANNEL PEAR.PHP.NET):
=====================================
Auto-discover new Channels     auto_discover    <not set>
Default Channel                default_channel  pear.php.net
HTTP Proxy Server Address      http_proxy       <not set>
PEAR server [DEPRECATED]       master_server    pear.php.net
Default Channel Mirror         preferred_mirror pear.php.net
Remote Configuration File      remote_config    <not set>
PEAR executables directory     bin_dir          /usr/bin
PEAR documentation directory   doc_dir          /usr/share/doc/pear
PHP extension directory        ext_dir          /usr/lib/php/modules
PEAR directory                 php_dir          /usr/share/pear
PEAR Installer cache directory cache_dir        /var/cache/php-pear
PEAR configuration file        cfg_dir          /etc/pear
directory
PEAR data directory            data_dir         /usr/share/pear-data
PEAR Installer download        download_dir     /opt/schoolserver/xsce
directory
PHP CLI/CGI binary             php_bin          /usr/bin/php
php.ini location               php_ini          <not set>
--program-prefix passed to     php_prefix       <not set>
PHP's ./configure
--program-suffix passed to     php_suffix       <not set>
PHP's ./configure
PEAR Installer temp directory  temp_dir         /var/tmp
PEAR test directory            test_dir         /usr/share/tests/pear
PEAR www files directory       www_dir          /var/www/html
Cache TimeToLive               cache_ttl        3600
Preferred Package State        preferred_state  stable
Unix file mask                 umask            22
Debug Log Level                verbose          1
PEAR password (for             password         <not set>
maintainers)
Signature Handling Program     sig_bin          /usr/bin/gpg
Signature Key Directory        sig_keydir       /etc/pearkeys
Signature Key Id               sig_keyid        <not set>
Package Signature Type         sig_type         gpg
PEAR username (for             username         <not set>
maintainers)
User Configuration File        Filename         /root/.pearrc
System Configuration File      Filename         /etc/pear.conf
```
Think you broke the compiler somehow, but as a test try to install by hand with:
```
pecl install zmq*tgz
Ignoring installed package pecl/zmq
Nothing to install
```
Now with -f
```
 pecl install -f zmq*tgz
12 source files, building
running: phpize
Configuring for:
PHP Api Version:         20100412
Zend Module Api No:      20100525
Zend Extension Api No:   220100525
Please provide the prefix of libzmq installation [autodetect] :    <<< just press enter here
building in /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2
running: /var/tmp/zmq/configure --with-zmq
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... yes
checking build system type... armv7l-unknown-linux-gnueabi
checking host system type... armv7l-unknown-linux-gnueabi
checking target system type... armv7l-unknown-linux-gnueabi
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib
checking for PHP extension directory... /usr/lib/php/modules
checking for PHP installed headers prefix... /usr/include/php
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking whether to enable 0MQ support... yes, shared
checking for pkg-config... /usr/bin/pkg-config
checking libzmq installation... found version 3.2.4, under /usr
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for stdint.h... (cached) yes
checking time.h usability... yes
checking time.h presence... yes
checking for time.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking mach/mach_time.h usability... no
checking mach/mach_time.h presence... no
checking for mach/mach_time.h... no
checking for library containing clock_gettime... -lrt
checking for clock_gettime... yes
checking for gettimeofday... yes
checking for mach_absolute_time... no
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /usr/bin/sed
checking for fgrep... /usr/bin/grep -F
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert armv7l-unknown-linux-gnueabi file names to armv7l-unknown-linux-gnueabi format... func_convert_file_noop
checking how to convert armv7l-unknown-linux-gnueabi file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... (cached) gawk
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for sysroot... no
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... no
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/usr/bin/ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating ./config.status
config.status: creating config.h
config.status: executing libtool commands
running: make
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=compile cc  -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/zmq/zmq.c -o zmq.lo
libtool: compile:  cc -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/zmq/zmq.c  -fPIC -DPIC -o .libs/zmq.o
/var/tmp/zmq/zmq.c: In function 'zim_zmqpoll_add':
/var/tmp/zmq/zmq.c:1153:3: warning: passing argument 2 of 'zend_throw_exception' discards 'const' qualifier from pointer target type [enabled by default]
In file included from /var/tmp/zmq/php_zmq_private.h:35:0,
                 from /var/tmp/zmq/zmq.c:32:
/usr/include/php/Zend/zend_exceptions.h:43:17: note: expected 'char *' but argument is of type 'const char *'
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=compile cc  -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/zmq/zmq_pollset.c -o zmq_pollset.lo
libtool: compile:  cc -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/zmq/zmq_pollset.c  -fPIC -DPIC -o .libs/zmq_pollset.o
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=compile cc  -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/zmq/zmq_device.c -o zmq_device.lo
libtool: compile:  cc -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/zmq/zmq_device.c  -fPIC -DPIC -o .libs/zmq_device.o
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=compile cc  -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/zmq/zmq_sockopt.c -o zmq_sockopt.lo
libtool: compile:  cc -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/zmq/zmq_sockopt.c  -fPIC -DPIC -o .libs/zmq_sockopt.o
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=compile cc  -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/zmq/zmq_fd_stream.c -o zmq_fd_stream.lo
libtool: compile:  cc -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/zmq/zmq_fd_stream.c  -fPIC -DPIC -o .libs/zmq_fd_stream.o
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=compile cc  -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /var/tmp/zmq/zmq_clock.c -o zmq_clock.lo
libtool: compile:  cc -I. -I/var/tmp/zmq -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /var/tmp/zmq/zmq_clock.c  -fPIC -DPIC -o .libs/zmq_clock.o
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=link cc -DPHP_ATOM_INC -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/include -I/var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/main -I/var/tmp/zmq -I/usr/include/php -I/usr/include/php/main -I/usr/include/php/TSRM -I/usr/include/php/Zend -I/usr/include/php/ext -I/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -o zmq.la -export-dynamic -avoid-version -prefer-pic -module -rpath /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/modules  zmq.lo zmq_pollset.lo zmq_device.lo zmq_sockopt.lo zmq_fd_stream.lo zmq_clock.lo -lzmq
libtool: link: cc -shared  -fPIC -DPIC  .libs/zmq.o .libs/zmq_pollset.o .libs/zmq_device.o .libs/zmq_sockopt.o .libs/zmq_fd_stream.o .libs/zmq_clock.o   -lzmq  -O2   -Wl,-soname -Wl,zmq.so -o .libs/zmq.so
libtool: link: ( cd ".libs" && rm -f "zmq.la" && ln -s "../zmq.la" "zmq.la" )
/bin/sh /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/libtool --mode=install cp ./zmq.la /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/modules
libtool: install: cp ./.libs/zmq.so /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/modules/zmq.so
libtool: install: cp ./.libs/zmq.lai /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/modules/zmq.la
libtool: finish: PATH="/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin:/root/bin:/sbin" ldconfig -n /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/modules
----------------------------------------------------------------------
Libraries have been installed in:
   /var/tmp/pear-build-rootiy7FIh/zmq-1.1.2/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.

running: make INSTALL_ROOT="/var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2" install
Installing shared extensions:     /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2/usr/lib/php/modules/
running: find "/var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2" | xargs ls -dils
13243670   4 drwxr-xr-x 3 root root   4096 Feb  5 16:18 /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2
13243701   4 drwxr-xr-x 3 root root   4096 Feb  5 16:18 /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2/usr
13243702   4 drwxr-xr-x 3 root root   4096 Feb  5 16:18 /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2/usr/lib
13243703   4 drwxr-xr-x 3 root root   4096 Feb  5 16:18 /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2/usr/lib/php
13243704   4 drwxr-xr-x 2 root root   4096 Feb  5 16:18 /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2/usr/lib/php/modules
13243700 244 -rwxr-xr-x 1 root root 248727 Feb  5 16:18 /var/tmp/pear-build-rootiy7FIh/install-zmq-1.1.2/usr/lib/php/modules/zmq.so

Build process completed successfully
Installing '/usr/lib/php/modules/zmq.so'
install ok: channel://pecl.php.net/zmq-1.1.2
configuration option "php_ini" is not set to php.ini location
You should add "extension=zmq.so" to php.ini

```

**Comment by tim-moody** [comment]
Could things go wrong in compile or link?
Maybe we should pecl build as a separate step.


Sent from my Samsung Galaxy smartphone.<div>
</div><div>
</div><div>-------- Original message --------</div><div>From: georgejhunt <notifications@github.com> </div><div>Date: 12/16/2015  12:34 PM  (GMT-05:00) </div><div>To: XSCE/xsce <xsce@noreply.github.com> </div><div>Cc: Tim Moody <tim@timmoody.com> </div><div>Subject: Re: [xsce] re-order gateway support (#538) </div><div>
</div>
Not sure this is relevant. I did a "pear config-show" and got a message
right at the top saying it could not open shared object
../php/modules/zmq.so.  So there is some memory/state in pear which we need
to clear out for rerunning our playbook.

When I try to run "pecl install -f zmq*beta" from the command line, I get

WARNING: channel "pecl.php.net" has updated its protocols, use "pecl
channel-update pecl.php.net" to update
downloading zmq-1.1.2.tgz ...
Starting to download zmq-1.1.2.tgz (39,573 bytes)
..........done: 39,573 bytes
12 source files, building
running: phpize
Configuring for:
PHP Api Version:         20100412
Zend Module Api No:      20100525
Zend Extension Api No:   220100525
Please provide the prefix of libzmq installation [autodetect] : building in
/var/tmp/pear-build-rootEzdFPW/zmq-1.1.2
running: /var/tmp/zmq/configure --with-zmq
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for a sed that does not truncate output... /usr/bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... configure: error: in
`/var/tmp/pear-build-rootEzdFPW/zmq-1.1.2':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
ERROR: `/var/tmp/zmq/configure --with-zmq' failed

the /var/tmp/pear-build disappears so it's hard to follow up. When I untar
the zmq*tgz, there's a config.m4, but no .configure, which I might be able
to run manually, and find out why it thinks we are cross-compiling.

On Wed, Dec 16, 2015 at 11:36 AM, Jerry Vonau <notifications@github.com>
wrote:

> On Wed, Dec 16, 2015 at 10:02 AM, Tim Moody <notifications@github.com>
> wrote:
>
> > is this doing an offline install?
> >
> I've mentioned this before, one can fake that by setting use_cache &
> no_network to True in local_vars.yml, but yes the F18 box is nothing but a
> LanController with no net access. Once a single pass through
> install-console is done the above overrides are effective in suppressing
> the downloads.
>
>
> > was download ever separated from install?
> >
> Yes, back in March so I'm at a bit of a loss as to why only now is this
> bubbling to the surface.
>
>
> > not sure version of python binding is relevant to absence of php binding
> > .so, but if the api didn't work it would be.
> >
> Just trying to narrow down why there is a change in behavior now, but the
> above change works with both F18 and F22 over here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/XSCE/xsce/pull/538#issuecomment-165167911>.
>


---

**Comment by tim-moody** [comment]
I thought -f did a fresh download

**Comment by jvonau** [comment]
check with pecl help install, but note we are installing from the tgz file
that was downloaded.

**Comment by tim-moody** [comment]
OK.  I missed the different behaviour depending on source

**Comment by georgejhunt** [comment]
I'd still rather see setting xsce_prepped (lines 31-35) at the end of main.yml, rather than close to the beginning, to address the erroring out during network install, and then never being able to complete the network install.

**Comment by tim-moody** [comment]
I think xsce installation does not properly distinguish between the case where an image from another machine is being used vs. the case where a fresh install is done with ansible or anaconda.  In the latter case it makes sense to preserve as much of the configuration as possible when testing the actual network discovered against the config.  In former it makes no sense as the image was prepared on a different machine. 

I wonder whether it would advance the cause to add an optional parameter 'reset' to xsce-network which would remove all  ifcfg- files (except lo) to make networking do a fresh install.  This could be used early on in the process of installing an image on a new machine or later manually if things get too strange.

**Comment by georgejhunt** [comment]
+1

**Comment by jvonau** [comment]
On Thu, Dec 17, 2015 at 1:41 PM, Tim Moody <notifications@github.com> wrote:

> I think xsce installation does not properly distinguish between the case
> where an image from another machine is being used vs. the case where a
> fresh install is done with ansible or anaconda.
>
It's all ansible, anaconda just calls the playbook. In reality it's tracked
if nobody deletes the xsce.ini file or re-run's install-console or
runansible. Note the recording of [version] is written in
0-once/tasks/xsce_ini.yml and [runtime] is done in
1-prep/tasks/computed_vars.yml. Take a look at the xsce.ini file on a
freshly booted liveiso, note [runtime] [version], now just running runtags
XXX should change [runtime] leaving [version] unaltered. We need to address
updating the cmd_srv files without needing to run install-console to update
a few files(runtags console?), and maybe record the git hash that
corresponds to the cmd_srv files in use?(think that could be useful from a
support POV to visually see if cmd_srv may need updating) I see value in
recording the entire history of the soup that is cooked up and served in a
bowl back to you to try. ;)

In the latter case it makes sense to preserve as much of the configuration
> as possible when testing the actual network discovered against the config.
> In former it makes no sense as the image was prepared on a different
> machine.
>
We should just blindly attempt to install rpms that are already installed?
That is part of networking also, it's a waste of cycles on limited resource
hardware.

> I wonder whether it would advance the cause to add an optional parameter
> 'reset' to xsce-network which would remove all ifcfg- files (except lo) to
> make networking do a fresh install.
>
We have, in effect, that in place right now with the deleting the
ifcfg-files within the routine now, what are you exactly asking for? I'd
rather work to make any routine work correctly all the time, so if you do
come across something that appears strange just post the log. Sorry I'm not
willing to work off of some comments like "it doesn't work", I would like
pictures and/or logs so I have a clearer picture in my head.


> This could be used early on in the process of installing an image on a new
> machine or later manually if things get too strange.
>
Thought that is what I did with the xsce-network bit...  Ok, maybe I'm
missing something, so what do you see as needing to cleared out between an
image and a virgin install? In order to debug this we need to really define
what the expected behavior should be 1) git clone on pre-existing os 2)
while building/altering images 3) after image is installed.

**Comment by tim-moody** [comment]
are you saying that all ifcfg files are now deleted in all cases?

where were 'some comments like "it doesn't work"'?  you must be referring to someone else.

**Comment by jvonau** [comment]
On Thu, Dec 17, 2015 at 3:44 PM, Tim Moody <notifications@github.com> wrote:

> are you saying that all ifcfg files are now deleted in all cases?
>
All LAN interfaces are *always* rebuild from the ground up, WAN depends on
a number of factors like available gateway, WAN via WiFi, gui's network
mode selector.

  > where were 'some comments like "it doesn't work"'? you must be
referring to someone else.

Generalization, wasn't aimed at anybody, just need some more information
beyond what I can gather from looking at a xsce.ini file, if I'm lucky
enough to see that, or being able to extract more info via questions. Think
about calling a Doctor on the phone saying "it hurts here" then the Dr
asking "where on your body are you point at?" That gets a bit monotonous on
a volunteer's time constraints is all I'm suggesting.

**Comment by jvonau** [comment]
Closing as I believe the root cause was elsewhere, could revisit if needed
